tests/test_job.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+"""Unit tests for job endpoints"""
+
+import json
+import pytest
+from src.rev_ai.apiclient import RevAiAPIClient
+from src.rev_ai.models import Job, JobStatus, JobSubmitOptions
+
+try:
+    from unittest.mock import patch, MagicMock
+    from urllib.parse import urljoin
+except ImportError:
+    from mock import patch, MagicMock
+    from urlparse import urljoin
+
+JOB_ID = '1'
+MEDIA_URL = "https://example.com/test.mp3"
+
+
+@pytest.mark.usefixtures("mockclient")
+class TestJobEndpoints():
+    def test_get_job_details_success(self, mockclient):
+        status = 'transcribed'
+        created_on = '2018-05-05T23:23:22.29Z'
+        data = {'id': JOB_ID, 'status': status, 'created_on': created_on}
+        mockclient.session.get.return_value.json.return_value = data
+
+        res = mockclient.get_job_details(JOB_ID)
+
+        assert res == Job(JOB_ID, created_on, JobStatus.TRANSCRIBED)
+        mockclient.session.get.assert_called_once_with(
+            urljoin(RevAiAPIClient.base_url, "jobs/{id}".format(id=JOB_ID))
+        )
+
+    def test_get_job_details_not_authorized_error(self, mockclient):
+        data = {
+            "title": "Authorization has been denied for this request",
+            "status": 401
+        }
+        mockclient.session.get.return_value.json.return_value = data
+
+        with pytest.raises(KeyError):
+            mockclient.get_job_details(JOB_ID)
+
+    def test_get_job_details_job_not_found_error(self, mockclient):
+        data = {
+            "type": "https://www.rev.ai/api/v1/errors/job-not-found",
+            "title": "could not find job",
+            "status": 404
+        }
+        mockclient.session.get.return_value.json.return_value = data
+
+        with pytest.raises(KeyError):
+            mockclient.get_job_details(JOB_ID)
+
+    def test_submit_job_url_success(self, mockclient):
+        status = 'in_progress'
+        created_on = '2018-05-05T23:23:22.29Z'
+        metadata = 'test'
+        data = {
+            'id': JOB_ID,
+            'status': status,
+            'created_on': created_on,
+            'metadata': metadata
+        }
+        mockclient.session.post.return_value.json.return_value = data
+
+        res = mockclient.submit_job_url(MEDIA_URL,
+                                        JobSubmitOptions(metadata=metadata))
```

Comment:
To be pedantic, do you want to also test the callback_url functionality (found it was missing via coverage command)?